lib/streamlit/elements/skeleton.py

```diff
@@ -0,0 +1,146 @@
+        render UI first and content second.
+
+        This means you can replace
+
+        >>> # Draw UI
+        >>> st.write("Census Data")
+        >>> census_data_placeholder = st.empty()  # The empty is a collapsed element
+        >>> # ^ The empty is a 0-size placeholder
+        >>> year = st.selectbox("Select Year", options=[2022, 2023, 2024])
+        >>>
+        >>> # Get data from source, assume this takes a long time
+        >>> census_dataframe = get_census_data_from_database(year)
+        >>>
+        >>> # Draw the data, causing it to "pop" in above the year selectbox.
+        >>> census_data_placeholder.dataframe(census_dataframe)
+
+        with
+
+        >>> # Draw UI
+        >>> st.write("Census Data")
+        >>> census_data_placeholder = st.skeleton(200)
+        >>> # ^ The skeleton holds "200pt" of space for future content
+        >>> year = st.selectbox("Select Year", options=[2022, 2023, 2024])
+        >>>
+        >>> # Get data from source, assume this takes a long time
+        >>> census_dataframe = get_census_data_from_database(year)
+        >>>
+        >>> # Draw the data, replacing the skeleton animation that was previously there
+        >>> census_data_placeholder.dataframe(census_dataframe)
+
+        Overwriting elements in-place using "with" notation:
+
+        >>> import streamlit as st
+        >>> import time
+        >>>
+        >>> with st.skeleton():
+        ...     for seconds in range(60):
+        ...         st.write(f"⏳ {seconds} seconds have passed")
+        ...         time.sleep(1)
+        ...     st.write("✔️ 1 minute over!")
+
+        Replacing several elements, then clearing them:
+
+        >>> import streamlit as st
+        >>>
+        >>> placeholder = st.skeleton()
+        >>>
+        >>> # Replace the placeholder with some text:
+        >>> placeholder.text("Hello")
+        >>>
+        >>> # Replace the text with a chart:
+        >>> placeholder.line_chart({"data": [1, 5, 2, 6]})
+        >>>
+        >>> # Replace the chart with several elements:
+        >>> with placeholder.container():
+        ...     st.write("This is one element")
+        ...     st.write("This is another")
+        ...
+        >>> # Clear all those elements and put the animation back:
+        >>> placeholder.skeleton()
+
+        Replacing a skeleton with several elements:
+
+        >>> import streamlit as st
+        >>>
+        >>> placeholder = st.skeleton()
+        >>>
+        >>> # Keep in mind the animation will be replaced
+        >>> # as soon as placeholder.container() is called.
+        >>> with placeholder.container():
+        ...     placeholder.write("Hello")
+        ...     placeholder.write("Streamlit")
+
+        """
+
+        skeleton_proto = SkeletonProto()
+
+        # Validate Inputs
+        if height is not None:
+            if height <= 0:
+                raise ValueError("Skeleton height must be a positive integer.")
```

Comment:
We are not super consistent about this, but we usually throw a `StreamlitAPIException` for any error related to the usage of the Streamlit API.